models/demos/t3000/llama2_70b/tt/llama_generation.py

```diff
@@ -50,47 +56,85 @@ def __init__(self, configuration, state_dict, model_args, tt_args, paged_attenti
             cache_path=tt_args.cache_path,
             read_cache=False,
             paged_attention_config=paged_attention_config,
+            vllm=vllm,
         )
 
         del state_dict
 
-    def forward(self, tokens: torch.Tensor, start_pos: int):
+    @classmethod
+    def initialize_vllm_model(cls, hf_config, t3k_mesh_device):
+        # TODO: pass in model args and tt args as parameters from vllm
+        @dataclass
+        class ModelArgs:
+            llama_version: str = None
+            ckpt_dir: str = None
+            max_batch_size: int = 32
+            num_layers: int = 80
+            max_kv_context_len: int = 4096
+
+        @dataclass
+        class TTArgs:
+            mesh_device: object = None
+            cache_path: str = None
+
+        # setup configs
+        llama_version = "llama3"
+        model_config, ckpt_dir, _, cache_path = setup_llama_env(
+            llama_version=llama_version,
+        )
+        check_mesh_device(t3k_mesh_device, model_config)
+
+        # initialize arg classes
+        model_args = ModelArgs(llama_version=llama_version, ckpt_dir=ckpt_dir)
+        tt_args = TTArgs(mesh_device=t3k_mesh_device, cache_path=cache_path)
+
+        # load state dict
+        state_dict = load_llama_state_dict(model_args.ckpt_dir, n_layers=model_args.num_layers)
+
+        # TODO: delete this configuration setup once llama can directly accept hf_config
+        from models.demos.t3000.llama2_70b.reference.llama.llama.model import ModelArgs as ReferenceModelArgs
+        from pathlib import Path
+        import json
```

Comment:
why do we import stuff in this function?

Reply:
@cglagovichTT Those imports can be moved to the top, my error

Reply:
I might leave this as-is since these imports are only required if vllm is initializing the model